LottieDropper/Dropbox/DropboxBrowserViewModel.m

```diff
@@ -92,19 +140,108 @@ - (void)listFolderContinueWithClient:(DBUserClient *)client cursor:(NSString *)c
 	 }];
 }
 
+#pragma mark: - Fetch local files
+
+/// Makes list of file names from the outputDirectory
+- (NSArray <NSString *> *) fetchFilenamesFromOutputDirectory {
+
+    NSError *error;
+    // TODO: filter out hidden filenames
+    NSArray <NSString*> * list = [self.fileManager contentsOfDirectoryAtPath:self.outputDirectory.path error:&error];
+	if (error == nil) {
+		return list;
+	} else {
+		NSLog(@"%@", error);
+		return  nil;
+	}
+}
+
+/// 1. Load local file names
+/// 2. Use file names to fill the fileDetails list
+-(void) setupFileDetailsFromLocalFilenames {
+	NSArray <NSString*> * localFilelist = [self fetchFilenamesFromOutputDirectory];
+
+	for (NSString * fileName in localFilelist) {
+		if ( ![fileName hasPrefix:@"."] && ![self fileDetailsContainsFileName:fileName]) {
+			[self.fileDetails addObject:[[DropboxDetailViewModel alloc] initWithFile:fileName client:self.client]];
+		}
+	}
+
+}
+
 #pragma mark: - File info
 
-- (NSArray<DropboxDetailViewModel *> *)fileDetails {
-	NSMutableArray <DropboxDetailViewModel *> *mapped = [NSMutableArray arrayWithCapacity:[self.entries count]];
-	[self.entries enumerateObjectsUsingBlock:^(DBFILESMetadata * file, NSUInteger idx, BOOL *stop) {
-		[mapped addObject: [[DropboxDetailViewModel alloc] initWithFile:file client:self.client]];
-	}];
+// 1. Iterates over all DBFILESMetadata received from Dropbox
+// 2. Makes fileDetails if file not already loaded from local calls
+// 3. Remove files not on dropbox
+// 4. Add empty files for all dropbox files
+- (void) mergeDropboxFileListWithFileDetails {
+    
+    // 1.
+    [self.dropboxFileCache enumerateObjectsUsingBlock:^(DBFILESMetadata * file, NSUInteger idx, BOOL *stop) {
+        // 2.
+        if (![self fileDetailsContainsFileName:file.name]) {
+            DropboxDetailViewModel * detail = [[DropboxDetailViewModel alloc] initWithFile:file.name client:self.client];
+            [self.fileDetails addObject: detail];
```

Comment:
💀Je hebt hier wel een memory leak je met voor de block

```swift
__weak DropBoxBrowserViewModel * weakSelf = self;
```

schrijven en dan in de block die weak self gebruiken